openssl/src/pkcs7.rs

```diff
@@ -146,6 +183,82 @@ impl Pkcs7 {
             .map(Pkcs7)
         }
     }
+
+    /// Set signed attributes in a PKCS#7 structure.
+    ///
+    /// `attributes` is a stack of the attributes to be added.
+    ///
+    #[corresponds(PKCS7_set_signed_attributes)]
+    pub fn set_signed_attributes(
+        signer_info: &Pkcs7SignerInfoRef,
+        attributes: Stack<X509Attribute>,
+    ) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::PKCS7_set_signed_attributes(
+                signer_info.as_ptr(),
+                attributes.as_ptr(),
+            ))?;
+            mem::forget(attributes);
+            Ok(())
+        }
+    }
+
+    /// Add a signed attribute to a PKCS7_SIGNER_INFO structure
+    ///
+    /// `nid` is the Nid of the attribute, `atrtype` is the attribute's type (an ASN.1 tag
+    /// value) and `value` is the ASN1 object to be added.
+    ///
+    //  Note: `value` is immutable, but the OpenSSL function takes a (non-const) void pointer. Thus,
+    //  we have to cast to mutable in the unsafe block.
+    //  OpenSSL takes ownership of `value`.
+    //
+    #[corresponds(PKCS7_add_signed_attribute)]
+    pub fn add_signed_attribute(
+        signer_info: &Pkcs7SignerInfoRef,
+        nid: Nid,
+        atrtype: Asn1Type,
```

Comment:
What happens if `atrtype` disagrees with the contents of `Asn1Object`?

Reply:
Well, inside openssl this comes down to
```C
void ASN1_TYPE_set(ASN1_TYPE *a, int type, void *value)
{
    if (a->type != V_ASN1_BOOLEAN
            && a->type != V_ASN1_NULL
            && a->value.ptr != NULL) {
        ASN1_TYPE **tmp_a = &a;
        asn1_primitive_free((ASN1_VALUE **)tmp_a, NULL, 0);
    }
    a->type = type;
    if (type == V_ASN1_BOOLEAN)
        a->value.boolean = value ? 0xff : 0;
    else
        a->value.ptr = value;
}
```
(https://github.com/openssl/openssl/blob/OpenSSL_1_1_1-stable/crypto/asn1/a_type.c#L26)
In the last line, all values are accepted without control. But I could check for `V_ASN1_BOOLEAN` and `V_ASN1_NULL`.
Is that, what you intended? Or should we check against all `V_ASN1_<Type>`?

Reply:
I am concerned about what the worst case outcome is of someone passing the wrong type into this function. Is it just weird attributes? Is it UB?